backend/api/Database/Models/RobotModel.cs

```diff
@@ -79,7 +79,7 @@ public void Update(UpdateRobotModelQuery updateQuery)
         /// Updates the <see cref="AverageDurationPerTag"/> based on the data in the <paramref name="recentMissionsForModelType"/> provided
         /// </summary>
         /// <param name="recentMissionsForModelType"></param>
-        public void UpdateAverageDurationPerTag(List<Mission> recentMissionsForModelType)
+        public void UpdateAverageDurationPerTag(List<MissionRun> recentMissionsForModelType)
         {
             if (recentMissionsForModelType.Any(mission => mission.Robot.Model.Type != Type))
```

Comment:
```suggestion
            if (recentMissionsForModelType.Any(missionRun => missionRun.Robot.Model.Type != Type))
```

Reply:
Go through the rest of the file and check this.

I would also to a search throughout the codebase for things named `mission`. It should be very clear now that we use either `missionRun` or `missionDefinition` as variable names. If we start using `mission` about both we will confuse ourselves.